spirv_reflect.c

```diff
@@ -2039,6 +2039,34 @@ static SpvReflectResult ParseDescriptorBindings(SpvReflectPrvParser* p_parser, S
     p_descriptor->type_description = p_type;
     p_descriptor->decoration_flags = ApplyDecorations(&p_node->decorations);
 
+    // Flags like non-writable and non-readable are found as member decorations only.
+    // If all members have one of those decorations set, promote the decoration up
+    // to the whole descriptor.
+    const SpvReflectPrvNode* p_type_node = FindNode(p_parser, p_type->id);
+    if (IsNotNull(p_type_node) && p_type_node->member_count) {
+      SpvReflectPrvDecorations common_flags = p_type_node->member_decorations[0];
+
+      for (uint32_t m = 1; m < p_type_node->member_count; ++m) {
```

Comment:
Was curious why this starts at 1 intead of 0?

Reply:
Member decoration at `0` is read in the line above. Then the rest are logical `&`ed one at a time. Result is if a decoration is `true` for all members, in the end it will remain `true`. These will be promoted to the whole descriptor. If there is only one decoration (at `0`) then its decorations will be promoted to the whole descriptor.